component/loki/process/internal/stages/pipeline.go

```diff
@@ -19,9 +19,17 @@ import (
 // StageConfig defines a single stage in a processing pipeline.
 // We define these as pointers types so we can use reflection to check that
 // exactly one is set.
+// TODO (@tpaschalis) The fact that this type belongs to an internal package
+// _and_ is part of the loki.process component Arguments, means that an
+// external caller will not be able to construct the component Arguments by
+// hand. This will be fixed once we've gained confidence in the ported
+// processing stages and the package is made non-internal.
```

Comment:
nit: please put TODO comments inside the struct definition as opposed to attached to the doc comments of StageConfig. Otherwise this will show up in pkg.go.dev which is always kind of weird to see from other packages IMO